src/sql/base/query/browser/untitledQueryEditorInput.ts

```diff
@@ -44,6 +48,31 @@ export class UntitledQueryEditorInput extends QueryEditorInput implements IUntit
 		return this.text.model.hasAssociatedFilePath;
 	}
 
+	override async save(group: GroupIdentifier, options?: ISaveOptions): Promise<IEditorInput | undefined> {
+		return this.createFileQueryEditorInput(group, options, true);
+	}
+
+	override async saveAs(group: GroupIdentifier, options?: ISaveOptions): Promise<IEditorInput | undefined> {
+		return this.createFileQueryEditorInput(group, options, false);
+	}
+
+	private async createFileQueryEditorInput(group: GroupIdentifier, options?: ISaveOptions, isRegularSave?: boolean): Promise<IEditorInput> {
+		// Create our own FileQueryEditorInput wrapper here so that the existing state (connection, results, etc) can be transferred from this input to the new file input.
+		let fileEditorInput: IEditorInput;
+		if (isRegularSave) {
```

Comment:
Why did you change this? What you had before was fine - moving it in here just means you have to add this extra logic instead of just passing in the editor input that each call makes itself.

Reply:
I thought moving it into one would work, but I'll put it back.